modules/imgproc/src/intersection.cpp

```diff
@@ -99,25 +99,43 @@ static int _rotatedRectangleIntersection( const RotatedRect& rect1, const Rotate
         vec2[i].y = pts2[(i+1)%4].y - pts2[i].y;
     }
 
+    //we adapt the epsilon to the smallest dimension of the rects
+    for( int i = 0; i < 4; i++ )
+    {
+        samePointEps = std::min(samePointEps, std::sqrt(vec1[i].x*vec1[i].x+vec1[i].y*vec1[i].y));
+        samePointEps = std::min(samePointEps, std::sqrt(vec2[i].x*vec2[i].x+vec2[i].y*vec2[i].y));
+    }
+    samePointEps = std::max(1e-16f, samePointEps);
+
     // Line test - test all line combos for intersection
     for( int i = 0; i < 4; i++ )
     {
         for( int j = 0; j < 4; j++ )
         {
             // Solve for 2x2 Ax=b
-            float x21 = pts2[j].x - pts1[i].x;
-            float y21 = pts2[j].y - pts1[i].y;
+            const float x21 = pts2[j].x - pts1[i].x;
+            const float y21 = pts2[j].y - pts1[i].y;
 
             float vx1 = vec1[i].x;
             float vy1 = vec1[i].y;
 
             float vx2 = vec2[j].x;
             float vy2 = vec2[j].y;
 
-            float det = vx2*vy1 - vx1*vy2;
+            float normalizationScale  = std::min(vx1*vx1+vy1*vy1, vx2*vx2+vy2*vy2);//sum of squares : this is >= 0
+            //normalizationScale is a square, and we usually limit accuracy around 1e-6, so normalizationScale should be rather limited by ((1e-6)^2)=1e-12
+            normalizationScale  = (normalizationScale < 1e-12f) ? 1.f : 1.f/normalizationScale;
+
+            vx1 *= normalizationScale;
+            vy1 *= normalizationScale;
+            vx2 *= normalizationScale;
+            vy2 *= normalizationScale;
+
+            const float det = vx2*vy1 - vx1*vy2;
+            const float detInvScaled = !det ? std::numeric_limits<float>::quiet_NaN() : (normalizationScale/det);
```

Comment:
> `!det`

Please avoid using bit-exact zero comparison with floating point variables. It could not work as expected.

Use bailout pre-check:

```
if (std::abs(det) < 1e-6)
    continue;
```